### ISSUE TITLE
Unicode confusables: Alif shouldn't be highlighted if between other Arabic letters

### ISSUE THREAD

**Issue by mustafa0x** [microsoft/vscode#140376]
![image](https://user-images.githubusercontent.com/1473102/148699342-b06b33c9-07fc-4f1d-971a-e27bd17a65cc.png)

```js
const str = `` // string removed since it incorrectly flagged the issue as not  being in english
```

It's a common Arabic letter. Maybe confusables should first check if the letter is surrounded by letters of the same alphabet. If it is, the warning is out of place.

**Comment by mustafa0x** [comment]
Well, now there's a meta bug: incorrectly assuming the description wasn't in English. :)

**Comment by IllusionMH** [comment]
There were more non-english text earlier in reports so it's not strange that automatic language detection picked it as primary language so not sure what improvement for bot can be here.

As for highlights.
Looks like this is part of https://code.visualstudio.com/updates/v1_63#_unicode-highlighting because letters might ambiguous characters. See provided settings on how to disable these warnings.

Looks like duplicate of #138905. Also see #138767 that should track OS locale and not warn.

**Comment by hediet** [comment]
Fixed by #140960

**Comment by mustafa0x** [comment]
> There were more non-english text earlier in reports so it's not strange that automatic language detection picked it as primary language so not sure what improvement for bot can be here.

Well it's not a big deal, but maybe the bot should disregard code blocks, as code blocks aren't descriptions, and that's what caused it to be flagged (even though the title and description below were in English).

@hediet Cool, thanks!
